src/components/arm/arm.ts

```diff
@@ -2,6 +2,8 @@ import type { JointPositions } from '../../gen/component/arm/v1/arm_pb';
 
 import type { Pose, Resource, StructType } from '../../types';
 
+export type ArmJointPositions = JointPositions;
```

Comment:
Can you not use JointPositions from the exported `armApi`?

https://github.com/viamrobotics/viam-typescript-sdk/blob/main/src/main.ts#L122

Reply:
That has a:
```
 @deprecated Use {@link ArmClient} instead.
 * @alpha
 * @group Raw Protobufs
```
comment. 

Should I remove that comment?

Reply:
Hah my bad. I'm not sure, I defer to the opinion of the SDK folks.

Reply:
Could we not just re-export `JointPositions`? Does it have to be renamed here? (pardon my ignorance...)

Reply:
@njooma 

Not sure -- the SDK is slightly inconsistent as to whether we rename before exporting. But I think it is mostly based on whether we expect the name to be re-used in another component. Because `JointPositions` is defined in the arm.proto, it seemed right to rename it with a prefix... ?

See:
https://github.com/viamrobotics/viam-typescript-sdk/blob/83a1bd0791d72618aab44bc98aedb819552dbd0c/src/components/movement-sensor/movement-sensor.ts
```
export type MovementSensorPosition = pb.GetPositionResponse.AsObject;
export type MovementSensorProperties = pb.GetPropertiesResponse.AsObject;
export type MovementSensorAccuracy = pb.GetAccuracyResponse.AsObject;
```
https://github.com/viamrobotics/viam-typescript-sdk/blob/83a1bd0791d72618aab44bc98aedb819552dbd0c/src/components/encoder/encoder.ts
```
export type EncoderProperties = pb.GetPropertiesResponse.AsObject;

type ValueOf<T> = T[keyof T];
export const EncoderPositionType = pb.PositionType;
export type EncoderPositionType = ValueOf<typeof pb.PositionType>;
```
But:
https://github.com/viamrobotics/viam-typescript-sdk/blob/83a1bd0791d72618aab44bc98aedb819552dbd0c/src/components/board/board.ts
```
export type Duration = PBDuration.AsObject;
export type AnalogValue = pb.ReadAnalogReaderResponse.AsObject;
```

Reply:
New to this SDK and haven't done SDK work in a while -- so if you think it is best to not rename, happy to remove that.

Reply:
RE: https://github.com/viamrobotics/viam-typescript-sdk/pull/316/files#r1651399656

This comment is still correct - if we provide a client wrapper then we mark the "native" proto client as deprecated, but still export it in case users have a need to work with the native client. At some point we can probably stop exporting native APIs, or more ideally starting using a different proto generator for JS...

RE: https://github.com/viamrobotics/viam-typescript-sdk/pull/316/files#r1653127091

We do want to export the custom type since it allows users to write type annotations. However, we generally convert it to a POJO with `.AsObject`. The whole idea is that users don't have to deal with native proto JS, just objects.

Reply:
Good point @maximpertsov . Added `asObject` in d27e425

Reply:
also any new types needs to be exported in `src/main.ts` to actually be importable from the package

Reply:
Oh -- my newness to this is showing -- I had assumed an export * above this. 7c16a9a

Reply:
no it's fine, this is an all-too-common oversight

Reply:
Are we worried that changing to return the `asObject` version is a breaking change?

Reply:
ah that's annoying, we were inconsistent with this type. @njooma I defer to you on this one - how do you feel about making this breaking change?

Reply:
(Commenting so I can see what decision we make)
From what I know, it's a pretty well-established pattern in the TS SDK that the responses are not the raw proto type but the objects. Any users who used this method had to treat this response differently than any other. But this means that future users will have to do this as well if we decide not to fix this.

Reply:
I think we should make this breaking change to bring things in line with all the other objects. Let's just make sure to announce it appropriately.

As for renaming -- I think we can rename to `ArmJointPositions` and, moving forward, make that the default -- if we're re-exporting a proto from a specific package, prefix with the package name. (So "common" protos wouldn't have a prefix, but specific component ones would).

Reply:
@njooma So just to confirm: you are okay with this PR moving forward as-is? If so, would you be willing to approve?

(I'm happy to make the eng-breaking-changes announcement unless you'd rather do it.)